File: 2970-count-the-number-of-incremovable-subarrays-i/2970-count-the-number-of-incremovable-subarrays-i.py
class Solution:
    def incremovableSubarrayCount(self, nums):
        count = 0
        length = len(nums)

        for i in range(length):
            for j in range(i, length):
                sub_arr = nums[i:j + 1] 
                temp = nums[:]
                temp[i:j + 1] = [] 

                strictly_increasing = True
                for k in range(len(temp) - 1):
                    if temp[k] >= temp[k + 1]:
                        strictly_increasing = False
                        break

                if strictly_increasing:
                    count += 1

        return count

File: 2970-count-the-number-of-incremovable-subarrays-i/2970-count-the-number-of-incremovable-subarrays-i.rs
impl Solution {
    pub fn incremovable_subarray_count(nums: Vec<i32>) -> i32 {
        let mut count = 0;
        let length = nums.len();

        for i in 0..length {
            for j in i..length {
                let sub_arr = &nums[i..=j];
                let mut temp = nums.clone();
                temp.splice(i..=j, [].iter().cloned());

                let strictly_increasing = temp.windows(2).all(|w| w[0] < w[1]);

                if strictly_increasing {
                    count += 1;
                }
            }
        }

        count
    }
}
